torch/_inductor/codegen/common.py

```diff
@@ -2024,7 +2025,24 @@ def do_cse(v):
                             bounds=bounds,
                             dtype=output_dtype,
                         )
+
+                        nonlocal idx
+                        if (
+                            config.test_configs.runtime_triton_dtype_assert
+                            and V.graph.get_current_device_or_throw().type == "cuda"
+                        ):
+                            from torch._inductor.codegen.triton import triton_type
+
+                            if isinstance(output_dtype, (list, tuple)):
+                                output_dtype = output_dtype[idx]
```

Comment:
Does this work because `do_cse` is called on each output in order by `pytree.tree_map`? It might be good to leave a comment explaining the logic since this is pretty tricky.